### PR TITLE
Draft implementation for Issue #204 (Swagger/Openapi: Multipart/form-data multiple file request support )

### PR DESCRIPTION
**Pull request by ydanneg**
Here is the draft PR to support Format specification of an array items.

This fixes https://github.com/bkbnio/kompendium/issues/204

### CODE REVIEW

File: kompendium-core/src/test/resources/formatted_array_item_type.json [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    "/test/formatted_type": {
+      "post": {
+        "tags": [],
+        "summary": "formatted array item type",
+        "description": "Cool stuff",
+        "parameters": [],
+        "requestBody": {
+          "description": "cool",
+          "content": {
+            "application/json": {
```
Comment (unredundant) [inline kompendium-core/src/test/resources/formatted_array_item_type.json:38]: I believe convention is to have the content type be `multipart/form-data`.  Would be good to capture that in the test. 

You should be able to do that by setting the `ResponseInfo.mediaTypes` field. Though admittedly... i've never actually tried it 😅 Just know I coded it up.

**Review by unredundant — COMMENTED** [review]
This looks great for the most part.  

I think this is blocked by #201 as mentioned in the comments.  

Once we resolve a way to support type variants (nullable, overriden format, etc. etc.) then we should be able to ship this right after

File: kompendium-core/src/main/kotlin/io/bkbn/kompendium/core/Kontent.kt [hunk 2]
```diff
@@ -81,7 +83,10 @@ object Kontent {
       Long::class -> cache[clazz.simpleName!!] = FormattedSchema("int64", "integer")
       Double::class -> cache[clazz.simpleName!!] = FormattedSchema("double", "number")
       Float::class -> cache[clazz.simpleName!!] = FormattedSchema("float", "number")
-      String::class -> cache[clazz.simpleName!!] = SimpleSchema("string")
+      String::class -> cache[clazz.simpleName!!] = SimpleSchema(
+        "string",
+        format = type.findAnnotation<Format>()?.format
+      )
```
[the remark this answers is not in the document]
Comment (ydanneg) [reply kompendium-core/src/main/kotlin/io/bkbn/kompendium/core/Kontent.kt:89]: oh.. you right... damn.. will re-think

[the remark this answers is not in the document]
Comment (unredundant) [reply kompendium-core/src/main/kotlin/io/bkbn/kompendium/core/Kontent.kt:89]: If you remove this, and then in the constraint scanner, check not just on the field but on the type, then we may be able to do this prior to fixing #201 outright.

[the remark this answers is not in the document]
Comment (ydanneg) [reply kompendium-core/src/main/kotlin/io/bkbn/kompendium/core/Kontent.kt:89]: > If you remove this, and then in the constraint scanner, check not just on the field but on the type, then we may be able to do this prior to fixing #201 outright.

wait.. but the result of constraint scanner still updates the cache...

**Comment by unredundant** [comment]
This looks great! Think you just need to resolve a conflict in the constraint scanner, add a line to the CHANGELOG, then we'll merge this and I'll cut a release branch later today :)

**Comment by ydanneg** [comment]
found an issue.. wait

**Comment by ydanneg** [comment]
would be great to cover scanner by tests.. but... let's wait for your big changes

**Pull request merged**